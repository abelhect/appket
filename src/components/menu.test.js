import React from 'react';
import {shallow} from 'enzyme';
import Menu from './menu';
import '../setupTests';


describe('All tests for Menu Component', () => {
  const menu = shallow(<Menu logo='logo'/>);

  //Test#1: (visual) rendering menu component
  it('renders correctly', () => {
    expect(menu).toMatchSnapshot();
  });

  //Test#2: () when you click on logo it sends you to home page

  //Test#3: () when you click on Log in it takes you to log in page

});
