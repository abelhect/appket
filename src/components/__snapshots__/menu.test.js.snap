// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All tests for Menu Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    logo="logo"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsStyle": "default",
      "children": Array [
        <NavbarHeader>
          <NavbarBrand>
                    <a
                              href="/home"
                    >
                              <strong>
                                        logo
                              </strong>
                    </a>
          </NavbarBrand>
          <NavbarToggle />
</NavbarHeader>,
        <NavbarCollapse>
          <Nav
                    bsClass="nav"
                    bsStyle="pills"
                    justified={false}
                    pullLeft={false}
                    pullRight={true}
                    stacked={false}
          >
                    <NavItem
                              active={false}
                              className="navitem"
                              disabled={false}
                              href="/home"
                    >
                              Log In
                    </NavItem>
          </Nav>
</NavbarCollapse>,
      ],
      "collapseOnSelect": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <NavbarBrand>
              <a
                            href="/home"
              >
                            <strong>
                                          logo
                            </strong>
              </a>
</NavbarBrand>,
            <NavbarToggle />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a
                href="/home"
>
                <strong>
                                logo
                </strong>
</a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <strong>
                  logo
</strong>,
                "href": "/home",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "logo",
                },
                "ref": null,
                "rendered": "logo",
                "type": "strong",
              },
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Nav
            bsClass="nav"
            bsStyle="pills"
            justified={false}
            pullLeft={false}
            pullRight={true}
            stacked={false}
>
            <NavItem
                        active={false}
                        className="navitem"
                        disabled={false}
                        href="/home"
            >
                        Log In
            </NavItem>
</Nav>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "nav",
            "bsStyle": "pills",
            "children": <NavItem
              active={false}
              className="navitem"
              disabled={false}
              href="/home"
>
              Log In
</NavItem>,
            "justified": false,
            "pullLeft": false,
            "pullRight": true,
            "stacked": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "active": false,
              "children": "Log In",
              "className": "navitem",
              "disabled": false,
              "href": "/home",
            },
            "ref": null,
            "rendered": "Log In",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsStyle": "default",
        "children": Array [
          <NavbarHeader>
            <NavbarBrand>
                        <a
                                    href="/home"
                        >
                                    <strong>
                                                logo
                                    </strong>
                        </a>
            </NavbarBrand>
            <NavbarToggle />
</NavbarHeader>,
          <NavbarCollapse>
            <Nav
                        bsClass="nav"
                        bsStyle="pills"
                        justified={false}
                        pullLeft={false}
                        pullRight={true}
                        stacked={false}
            >
                        <NavItem
                                    active={false}
                                    className="navitem"
                                    disabled={false}
                                    href="/home"
                        >
                                    Log In
                        </NavItem>
            </Nav>
</NavbarCollapse>,
        ],
        "collapseOnSelect": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <NavbarBrand>
                <a
                                href="/home"
                >
                                <strong>
                                                logo
                                </strong>
                </a>
</NavbarBrand>,
              <NavbarToggle />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a
                  href="/home"
>
                  <strong>
                                    logo
                  </strong>
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <strong>
                    logo
</strong>,
                  "href": "/home",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "logo",
                  },
                  "ref": null,
                  "rendered": "logo",
                  "type": "strong",
                },
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Nav
              bsClass="nav"
              bsStyle="pills"
              justified={false}
              pullLeft={false}
              pullRight={true}
              stacked={false}
>
              <NavItem
                            active={false}
                            className="navitem"
                            disabled={false}
                            href="/home"
              >
                            Log In
              </NavItem>
</Nav>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "nav",
              "bsStyle": "pills",
              "children": <NavItem
                active={false}
                className="navitem"
                disabled={false}
                href="/home"
>
                Log In
</NavItem>,
              "justified": false,
              "pullLeft": false,
              "pullRight": true,
              "stacked": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "active": false,
                "children": "Log In",
                "className": "navitem",
                "disabled": false,
                "href": "/home",
              },
              "ref": null,
              "rendered": "Log In",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
